lib/MusicBrainz/Server/View/Node.pm

```diff
@@ -38,20 +39,28 @@ sub process {
             sessionid => scalar($c->sessionid),
             session => $c->session,
             flash => $c->flash,
-        }});
+        },
+    };
 
     my $uri = URI->new;
     $uri->scheme('http');
     $uri->host(DBDefs->RENDERER_HOST || '127.0.0.1');
     $uri->port(DBDefs->RENDERER_PORT);
     $uri->path($c->req->path);
+    $uri->query_param_append('token', generate_token());
+    $uri->query_param_append('user', $c->user->name) if $c->user_exists;
```

Comment:
Umm. At the moment, of course, all this is only used for the 404 page, but in theory, one day, it will be called for every single website request. I see a risk that, at that point, the kernel entropy pool could be depleted rather quickly by this.